components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Storage/PublishedNodesJobConverterTests.cs

```diff
@@ -1253,6 +1253,258 @@ private class StandaloneIdentity : IIdentity {
                     dataSetWriter.DataSet.DataSetSource.PublishedVariables.PublishedData.Count));
         }
 
+        [Fact]
+        public void PnPlcEventTest() {
+            var pn = new StringBuilder(@"
+[
+    {
+        ""EndpointUrl"": ""opc.tcp://localhost:50000"",
+        ""OpcEvents"": [
+           {
+                ""Id"": ""i=2258"",
+                ""EventFilter"": {
+                    ""SelectClauses"": [
+                        {
+                            ""TypeId"": ""i=2041"",
+                            ""BrowsePaths"": [
+                            ""EventId""
+                            ]
+                        }
+                    ],
+                   ""WhereClause"": {
+                        ""Elements"": [
+                            {
+                                ""Operator"": ""OfType"",
+                                ""Operands"": [
+                                    {
+                                        ""Attribute"": {
+                                            ""NodeId"": ""ns=2;i=235"",
+                                            ""BrowsePath"": """"
+                                            }
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                }
+            }
+        ]
+    }
+]
+");
+
+            var converter = new PublishedNodesJobConverter(TraceLogger.Create(), _serializer);
+            var jobs = converter.Read(new StringReader(pn.ToString()), new LegacyCliModel()).ToList();
+
+            Assert.Single(jobs);
+            Assert.Single(jobs.Single().WriterGroup.DataSetWriters);
+            Assert.Single(jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedEvents.PublishedData);
+            Assert.NotNull(jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedVariables);
+            Assert.NotNull(jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedEvents);
+            Assert.Empty(jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedVariables.PublishedData);
+            Assert.Single(jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedEvents.PublishedData);
+
+            var model = jobs.Single().WriterGroup.DataSetWriters.Single().DataSet.DataSetSource.PublishedEvents.PublishedData.Single();
+            Assert.Equal("i=2258", model.Id);
+
+            // TODO: ...
```

Comment:
please check, PR should have solved all TODO

Reply:
maybe checking for Operator, Operands, BrowsePaths etc.

---
In reply to: [596731348](https://github.com/Azure/Industrial-IoT/pull/1063#discussion_r596731348) [](ancestors = 596731348)

Reply:
Checking with @Dag, but I agree.

Reply:
I'll take this and fix it.

Reply:
Fixed.